### ISSUE TITLE
Decide on and implement uniform approach to normative keywords

### ISSUE THREAD

**Issue by lgolding** [oasis-tcs/sarif-spec#78]
We adhere to RFC 2119 in that we distinguish three categories of normative statements:
- The statement is an absolute requirement.
- There might be a valid reason to ignore the statement.
- The statement is truly optional.

We also adhere to RFC 2119 in that we use the (capitalized) keywords SHALL and MUST to designate "absolute requirements." (RFC 2119 also allows "REQUIRED", but we don't currently use it.) We treat SHALL and MUST as absolutely synonymous, and we choose between them on a case by case basis, purely on the basis of which one reads most naturally in English.

Stefan and Ykaterina have both said that they find it easier to distinguish MUST from SHOULD than to distinguish SHALL from SHOULD, and Stefan has suggested using MUST exclusively. On the other hand, if we want to be ISO ready, we should not use MUST in this sense, since "must" means something else in an ISO spec. On the other other hand, Stefan cites the OData spec, which was written to OASIS standards, and then accepted without change by ISO.

We need to establish, and then revise the spec to conform to, a policy for keyword usage that meets the following requirements:
- Maximize understandability for both native English speakers and non-native speakers.
- Maximize consistency of expression throughout the spec (this improves understandability _except_ in cases where an arbitrarily imposed consistency leads to strained expression).
- Make the spec ISO ready (_if_ the TC agrees that this is a goal, which it has not yet explicitly done).

**Comment by lgolding** [comment]
Here is a proposal, suggested in email by David Keaton, and amplified by me with implementation details:

We will adopt the common subset of RFC 2219/8174 and the ISO Directive. That is, the spec will be fully compliant with both, so that anyone who is used to reading documents in one format or the other will be comfortable with our spec, because they won't have to learn anything new to read it. These benefits apply whether or not we ever submit the spec to ISO, and this approach increases our options for the future.

In particular, we will do the following:
1. Capitalize normative keywords. We do this already.
2. Use SHALL exclusively (not MUST). There are only a few occurrences of normative "MUST" in the spec, so this is easy.
3. Do not use lower case versions of the normative keywords with their "common English" meanings, because ISO allows keywords with either casing, so readers might mistake such language as normative. Instead, use the synonyms suggested in the OASIS keyword guidelines. This is the largest change: there are many places in the spec where lower case keywords are used non-normatively.

**Comment by michaelcfanning** [comment]
Amend bullet 3 to replace 'Do not' with avoid.

**Comment by michaelcfanning** [comment]
We decided not to overinvest in worrying about #3.

**Comment by michaelcfanning** [comment]
Promote to 'editor's design guidelines document'

**Comment by lgolding** [comment]
@michaelcfanning Right, we actually changed #3 to be "Avoid" rather than "do not" so I wouldn't have to scrub for this. This is only a few minutes work. I will do this once we merge the changes from tomorrow's TC meeting, and then close it.

**Comment by michaelcfanning** [comment]
@lgolding is this work done?

**Comment by lgolding** [comment]
@michaelcfanning No. I was going to do it after I finished merging all of last week's changes (which I've now done); instead I forgot and started on new change drafts for remaining issues. I'll have it done by next meeting.

**Comment by lgolding** [comment]
@michaelcfanning Working on this now. Not in numeric order with the rest of the CSD.1 issues, but easiest to do before adding more new content.

**Comment by lgolding** [comment]
@michaelcfanning This is done. I scrubbed for every occurrence of a lower-case normative keyword. Several of them _should have_ been normative, and I bold-capped those. Some of them had easy obvious rephrasings, and I did those. I _ignored_ all occurrences in non-normative text (examples, notes, and "informative" appendices).

After all that, there were only about three occurrences of lower-case normative keywords in normative text that it would have taken me a bit longer to rephrase, so I left them.

The entire process (including updating the Editor's Report, sending TC mail, and writing this comment) took 45 minutes. With that in mind, I'll be happy to look at those last few occurrences if you want :-)
